third_party/amd/lib/TritonAMDGPUToLLVM/DotOpToLLVM/MFMA.cpp

```diff
@@ -724,9 +766,18 @@ LogicalResult convertScaledMFMA(triton::DotScaledOp op,
          isa<LinearEncodingAttr>(op.getB().getType().getEncoding()) &&
          "Both lhs and rhs should be linear layout.");
 
-  assert(isa<LinearEncodingAttr>(op.getAScale().getType().getEncoding()) &&
-         isa<LinearEncodingAttr>(op.getBScale().getType().getEncoding()) &&
-         "Both LhsScale and RhsScale should be linear layout.");
+  auto aScale = op.getAScale();
+  auto bScale = op.getBScale();
+
+  assert(((aScale && bScale) || (!aScale && !bScale)) &&
```

Comment:
This can be confusing that it can reads that we don't support mixed scaled and non-scaled cases. Can you add some comment to explain that when we lower normal `tt.dot` to `DotScaledOp` we will provide null value for both scales; when we lower `tt.dot_scaled` to `DotScaleOp` we will provide both scale values, even if one operand does not have associated scale, which then we will provide a constant one scale?

Reply:
Sure. Besides, there'll be a follow-up PR to avoid creating constant scale for `tt.dot_scaled` if both scales don't exist.